### PR TITLE
doc: entropy_cc310 docupdate

### PR DESCRIPTION
**Pull request by FrancescoSer**
Added mention to nrf5340.
Reviewed and edited the text on the page.

Signed-off-by: Francesco Servidio <francesco.servidio@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/drivers/entropy_cc310.rst [hunk 1]
```diff
@@ -7,18 +7,19 @@ CC310 entropy driver
...
-This hardware is available on the nRF52840 and nRF9160 devices.
+You can use the CC3xx entropy driver (*entropy_cc310*) to generate random data using the Arm CryptoCell CC310 hardware available on the following Nordic SoCs:
 
-When used on nRF52840, the entropy_cc3xx driver gathers entropy by using the CC310 hardware through the :ref:`nrf_cc3xx_platform_readme`.
+* nRF52840
+* nRF5340
+* nRF9160
 
-When used on nRF9160, the driver behavior depends on whether it is used in a secure or non-secure application:
+When used in conjunction with the nRF52840, the entropy_cc3xx driver gathers entropy by using the CC310 hardware through the :ref:`nrf_cc3xx_platform_readme`.
```
Comment (peknis) [inline doc/nrf/drivers/entropy_cc310.rst:16]: - I would rather say: When used with nRFxxx
- I believe we don't need the definite article in front of the product name -> nRF52840 etc.

File: doc/nrf/drivers/entropy_cc310.rst [hunk 2]
```diff
@@ -7,18 +7,19 @@ CC310 entropy driver
...
+When used in conjunction with the nRF5340 or nRF9160, the driver behavior depends on whether it is used in a secure or non-secure application:
 
-When used on nRF5340, the driver behavior depends on whether it is used in a secure or non-secure application:
+* In a secure application, or when the :ref:`lib_spm` is not used, the entropy_cc3xx driver gathers entropy by using the CC3xx hardware through the :ref:`nrf_cc3xx_platform_readme`.
```
Comment (peknis) [inline doc/nrf/drivers/entropy_cc310.rst:19]: Not sure about this, but should we have entropy_cc3xx between double backticks to highlight it? Or then have italics as you used that above?

**Review by peknis — CHANGES_REQUESTED** [review]

File: doc/nrf/drivers/entropy_cc310.rst (hunk 2 above)
Comment (greg-fer) [reply doc/nrf/drivers/entropy_cc310.rst:19]: I would rather use the complete name of the library, as in the title: CC310 entropy driver.
We have some defined rules for highlighting content (https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/doc_styleguide.html) and it looks like the emphasis in line 10 would be fine, but in other occurrences we should just use the library name?

Also, I see inconsistency with the CC310 in the title that later becomes CC3xx... Why is this?

File: doc/nrf/drivers/entropy_cc310.rst [hunk 3]
```diff
@@ -7,18 +7,19 @@ CC310 entropy driver
...
+* In a non-secure application, the driver gathers entropy through the :ref:`lib_secure_services` library.
 
-* When used :ref:`in a secure application <ug_nrf9160>`, or when the :ref:`lib_spm` is not used, the entropy_cc3xx driver gathers entropy by using the CC3xx hardware through the :ref:`nrf_cc3xx_platform_readme`.
-
-* When used :ref:`in a non-secure application <ug_nrf9160>`, the driver gathers entropy through the :ref:`lib_secure_services` library.
+For more details, see :ref:`ug_nrf5340` and :ref:`ug_nrf9160`.
```
Comment (greg-fer) [inline doc/nrf/drivers/entropy_cc310.rst:22]: These are pretty generic references, and looking for entropy in both gives no results. Can the link be more specific?

File: doc/nrf/drivers/entropy_cc310.rst [hunk 4]
```diff
@@ -7,18 +7,19 @@ CC310 entropy driver
...
    :depth: 2
 
-You can use the CC3xx entropy driver (entropy_cc310) to generate random data using the Arm CryptoCell CC310 hardware.
-This hardware is available on the nRF52840 and nRF9160 devices.
+You can use the CC3xx entropy driver (*entropy_cc310*) to generate random data using the Arm CryptoCell CC310 hardware available on the following Nordic SoCs:
 
-When used on nRF52840, the entropy_cc3xx driver gathers entropy by using the CC310 hardware through the :ref:`nrf_cc3xx_platform_readme`.
+* nRF52840
+* nRF5340
+* nRF9160
```
Comment (greg-fer) [inline doc/nrf/drivers/entropy_cc310.rst:14]: I'm tempted to suggest using sample_board_rows here, but it might be a bit too much information...

**Comment by FrancescoSer** [comment]
@greg-fer @peknis Implemented most of your suggestions.

**Review by greg-fer — APPROVED** [review]

**Review by peknis — APPROVED** [review]

**Review by nvlsianpu — APPROVED** [review]

**Pull request merged**